conan_provider.cmake

```diff
@@ -451,8 +451,46 @@ function(conan_install)
     # success
     set_property(GLOBAL PROPERTY CONAN_INSTALL_SUCCESS TRUE)
 
+
+    conan_set_program_path()
+
+    list(PREPEND CMAKE_PROGRAM_PATH ${_CONAN_PROGRAM_PATH})
+    set(CMAKE_PROGRAM_PATH ${CMAKE_PROGRAM_PATH} PARENT_SCOPE)
+
 endfunction()
 
+macro(conan_set_program_path)
+    set(_CONAN_PROGRAM_PATH "")
+
+    message("${conan_stdout}")
+    string(JSON _conan_graph_nodes_length LENGTH ${conan_stdout} graph nodes)
+    math(EXPR _conan_graph_nodes_length "${_conan_graph_nodes_length} - 1")
+    foreach(_node_number RANGE ${_conan_graph_nodes_length})
+
+        string(JSON _dep_binary GET ${conan_stdout} graph nodes ${_node_number} binary)
+        string(JSON _dep_context GET ${conan_stdout} graph nodes ${_node_number} context)
+        
+        if(NOT _dep_binary STREQUAL "Cache" OR NOT _dep_context STREQUAL "build")
```

Comment:
"Downloaded" or "Updated" will also be available to use, why ignoring them?

Reply:
Oh I see! Yeah I need to add those - the ones that have "skip" still have the fields but they are null, so I need a way to filter out anything that isn't valid. Perhaps this isn't the best field to check if the binary package _is_ in the cache (I'm assuming that Cache, Download _and_ Updated for the `binary` field _ALL_ mean they are in the cache with valid paths?)

Reply:
Isn't it enough to filter out the ones with ``binary=="Skip"``? that should work I think.

Reply:
mmm what would happen if we have different versions of the same dependency (via tool requires), and `tools.graph:skip_binaries=False` ? would there be a chance that we add `bin` dirs for different versions of the samae dependency in this case?

Reply:
You can check the node ``context`` attribute, it will be "host" and "build", you can differentiate it.

Reply:
Can go with the condition "binary is not skip" and "context is build" - hopefully that covers all cases correctly!

Reply:
I think so